### PR TITLE
Pull request

### PR DESCRIPTION
**Pull request by ProfliX**
Removed redundant call toString()
Optimized imports
Rearanged/refactored code (for better readability)
Val(same as var but with final) and var (lombok) usement for whole project (where is actually needed).
Unused code suppressed
Removed Map,Collection,List raw types
FileUtils > extractResource() > Asserted not null inputStream
DataStorage > getEnum > assertet object instanceof String;
Added JavaDocs to ConfigSettings and ReloadSettings
LightningSerializer > deserialize improvement
LightningException > Removed NotNull annotation from primitive type members.
Bumped new version > 3.2.4

### CODE REVIEW

File: src/main/java/de/leonhard/storage/Json.java [hunk 1]
```diff
@@ -38,18 +38,27 @@ public Json(final String name, final String path, final InputStream inputStream)
...
+            FileUtils.writeToFile(this.file, inputStream);
+        }
+        if (reloadSettings != null) {
+            this.reloadSettings = reloadSettings;
+        }
         forceReload();
     }
 
-    public Json(final File file) {
+    public Json(final File file)
```
Comment (KotlinFactory) [inline src/main/java/de/leonhard/storage/Json.java:73]: This bracket should not be there.

File: src/main/java/de/leonhard/storage/Json.java [hunk 2]
```diff
@@ -66,7 +75,8 @@ public Json(final File file) {
      * @return Map
      */
     @Override
-    public final Map<?, ?> getMap(final String key) {
+    public final Map<?, ?> getMap(final String key)
+    {
```
Comment (KotlinFactory) [inline src/main/java/de/leonhard/storage/Json.java:92]: Same for here -> Bracket to Line 78

File: src/main/java/de/leonhard/storage/Toml.java [hunk 3]
```diff
@@ -31,14 +32,18 @@ public Toml(final String name, final String path, final InputStream inputStream)
         this(name, path, inputStream, null, null);
     }
 
-    public Toml(@NonNull final String name, @NonNull final String path, @Nullable final InputStream inputStream, @Nullable final ReloadSettings reloadSettings, @Nullable final Consumer<FlatFile> reloadConsumer) {
+    public Toml(@NonNull final String name, @NonNull final String path,
```
Comment (KotlinFactory) [inline src/main/java/de/leonhard/storage/Toml.java]: This is way better than it was before. Thanks!

File: src/main/java/de/leonhard/storage/Yaml.java [hunk 4]
```diff
@@ -95,7 +110,13 @@ public Yaml addDefaultsFromInputStream(@Nullable final InputStream inputStream)
 
             val newData = new FileData(data, DataType.UNSORTED);
 
-            for (val key : newData.keySet()) if (!this.fileData.containsKey(key)) this.fileData.insert(key, newData.get(key));
+            for (val key : newData.keySet())
+            {
```
Comment (KotlinFactory) [inline src/main/java/de/leonhard/storage/Yaml.java:117]: Same for here. This is not formatting that is used in Java

**Review by KotlinFactory — CHANGES_REQUESTED** [review]
Some of the changes are great but brackets belong on the same line. (This is general and not only for the comments I added.)

File: src/main/java/de/leonhard/storage/Json.java (hunk 1 above)
Comment (ProfliX) [reply src/main/java/de/leonhard/storage/Json.java:73]: Okay.

**Comment by ProfliX** [comment]
By the way.. somewhere i really gotta use whitespace bracket style (like for methods with much parameters)

**Pull request closed without merging**